File: converted/classes_only/classes_only/PZXmlUtil$UnmarshallerAllocator.java
package zombie.util;

import java.util.HashMap;
import java.util.Map;
import javax.xml.bind.JAXBContext;
import javax.xml.bind.JAXBException;
import javax.xml.bind.Unmarshaller;
import zombie.util.PZXmlUtil.UnmarshallerAllocator.1;

final class PZXmlUtil$UnmarshallerAllocator {
   private static final ThreadLocal<PZXmlUtil$UnmarshallerAllocator> instance = ThreadLocal.withInitial(PZXmlUtil$UnmarshallerAllocator::new);
   private final Map<Class, Unmarshaller> m_map = new HashMap<>();

   private PZXmlUtil$UnmarshallerAllocator() {
   }

   public static <T> Unmarshaller get(Class<T> var0) throws JAXBException {
      return instance.get().getOrCreate(var0);
   }

   private <T> Unmarshaller getOrCreate(Class<T> var1) throws JAXBException {
      Unmarshaller var2 = this.m_map.get(var1);
      if (var2 == null) {
         JAXBContext var3 = JAXBContext.newInstance(new Class[]{var1});
         var2 = var3.createUnmarshaller();
         var2.setListener(new 1(this));
         this.m_map.put(var1, var2);
      }

      return var2;
   }
}

File: DoNotUse/include/zombie/util/PZXmlUtil$UnmarshallerAllocator.cpp
#pragma once
#include <string>
#include <vector>
#include <memory>
#include <unordered_map>
#include <unordered_set>
#include <cstdint>
#include "javax/xml/bind/JAXBContext.h"
#include "javax/xml/bind/JAXBException.h"
#include "javax/xml/bind/Unmarshaller.h"
#include "zombie/util/PZXmlUtil/UnmarshallerAllocator/1.h"

namespace zombie {
namespace util {


class PZXmlUtil {
public:
   private static const ThreadLocal<PZXmlUtil$UnmarshallerAllocator> instance = ThreadLocal.withInitial(PZXmlUtil$UnmarshallerAllocator::new);
   private const Map<Class, Unmarshaller> m_map = std::make_unique<std::unordered_map<>>();

   private PZXmlUtil$UnmarshallerAllocator() {
   }

   public static <T> Unmarshaller get(Class<T> var0) throws JAXBException {
      return instance.get().getOrCreate(var0);
   }

   private <T> Unmarshaller getOrCreate(Class<T> var1) throws JAXBException {
    Unmarshaller var2 = this.m_map.get(var1);
      if (var2 == nullptr) {
    JAXBContext var3 = JAXBContext.newInstance(new Class[]{var1});
         var2 = var3.createUnmarshaller();
         var2.setListener(std::make_shared<1>(this));
         this.m_map.put(var1, var2);
      }

    return var2;
   }
}
} // namespace util
} // namespace zombie
